wallet/src/actors/worker/methods.rs

```diff
@@ -283,4 +295,175 @@ impl Worker {
+        let req = types::RpcRequest::method(method)
+            .timeout(self.node.requests_timeout)
+            .params(params)
+            .expect("params failed serialization");
+
+        let f = self
+            .node
+            .address
+            .send(req)
+            .flatten()
+            .map(|json| {
+                log::debug!("getBlockChain request result: {:?}", json);
+                match serde_json::from_value::<Vec<types::ChainEntry>>(json).map_err(node_error) {
+                    Ok(blocks) => Ok(blocks),
+                    Err(e) => Err(e),
+                }
+            })
+            .flatten()
+            .map_err(|err| {
+                log::warn!("getBlockChain request failed: {}", &err);
+                err
+            });
+
+        Compat01As03::new(f).await
+    }
+
+    /// Ask a Witnet node for the contents of a single block.
+    pub async fn get_block(&self, block_id: String) -> Result<types::ChainBlock> {
+        log::debug!("Getting block with id {} ", block_id);
+        let method = String::from("getBlock");
+        let params = block_id;
+
+        let req = types::RpcRequest::method(method)
+            .timeout(self.node.requests_timeout)
+            .params(params)
+            .expect("params failed serialization");
+        let f = self
+            .node
+            .address
+            .send(req)
+            .flatten()
+            .map(|json| {
+                log::debug!("getBlock request result: {:?}", json);
+                serde_json::from_value::<types::ChainBlock>(json).map_err(node_error)
+            })
+            .flatten()
+            .map_err(|err| {
+                log::warn!("getBlock request failed: {}", &err);
+                err
+            });
+
+        Compat01As03::new(f).await
+    }
+
+    pub async fn handle_block(
+        &self,
+        block: types::ChainBlock,
+        wallet_id: &str,
+        wallet: types::SessionWallet,
+    ) -> Result<()> {
+        // NOTE: Possible enhancement.
+        // Maybe is a good idea to use a shared reference Arc instead of cloning this vector of txns
+        // if this vector results to be too big, problem is that doing so conflicts with the internal
+        // Cell of the txns type which cannot be shared between threads.
+        let block_epoch = block.block_header.beacon.checkpoint;
+        let block_hash = block.hash().as_ref().to_vec();
+
+        // Block transactions to be indexed.
+        // NOTE: only `VttTransaction` and `DRTransaction` are currently supported.
+        let dr_txns = block
+            .txns
+            .data_request_txns
+            .into_iter()
+            .map(types::Transaction::from);
+        let vtt_txns = block
+            .txns
+            .value_transfer_txns
+            .into_iter()
+            .map(types::Transaction::from);
+        let block_txns = dr_txns.chain(vtt_txns).collect::<Vec<types::Transaction>>();
+
+        for slf in &self.own_address {
```

Comment:
Iterating over an option

```suggestion
        if let Some(slf) = self.own_address.as_ref() {
```

Reply:
![image](https://user-images.githubusercontent.com/1316439/77756098-d3a78880-702e-11ea-8394-390b79b9ff46.png)

Reply:
Anyway, why is this an if? If `self.own_address` is `None`, this method will do nothing, so can't we just panic?

Reply:
Agree.

Anyhow, the two last commit remove the need for worker to know their own address altogether.